model/src/main/proto/interaction_object.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Object type of the interaction object
+enum ObjectType {
+  NORMALIZED_STRING = 0;
+  SIGNED_INT = 1;
+  NON_NEGATIVE_INT = 2;
+  REAL = 3;
+  BOOLEAN = 4;
+  NUMBER_WITH_UNITS = 5;
+  SET_OF_HTML_STRING = 6;
+}
+
+// Structure for any interaction object
+message InteractionObject {
+  ObjectType object_type = 1;
+  oneof value {
+    string normalized_string = 2;
+    int32 signed_int = 3;
+    uint32 non_negative_int = 4;
+    float real = 5;
+    bool bool_value = 6;
+    NumberWithUnits number_with_units = 7;
+    // repeated fields cannot be within a oneof so lists are
+    // wrapped into this parameter.
+    List set_of_html_string = 8;
+  }
+}
+
+// Structure containing lists. Note we should fill only
+// one of the fields in this message.
+// We do not add a oneof here because repeated fields cannot be within a oneof.
+message List {
+  repeated string string_list = 1;
+}
+
+// Structure for a number with units object.
+message NumberWithUnits {
+  ObjectType number_type = 1;
```

Comment:
Could this maybe instead be a oneof between the float & Fraction? I don't think we'd ever have a NumberWithUnits containing a list of strings, for instance.

Reply:
Done